### DISCUSSION TITLE
Average Colour Missing GB (of RGB) Components for Grey Images

### DISCUSSION THREAD

**Discussion by uriel-tannen** [ImageMagick/ImageMagick#3031] [Help] [answered]
I'm using the following code to calculate the average RGB values of an image:

`magick $filename -format "%f,%[fx:int(mean.r*255)],%[fx:int(mean.g*255)],%[fx:int(mean.b*255)]\n" info: >> "swatchesRGB.csv"`

However I've noticed that when the average colour is a pure grey colour (i.e. when R=G=B) the output has the correct value for R but leaves G and B as 0. 

For example running the above on this image outputs RGB(93,0,0) when the correct value should be RGB(93,93,93).

![Swatch](https://user-images.githubusercontent.com/36728824/103111028-5900ad80-4640-11eb-9104-85f51b88279a.jpg)

I'm keen to understand why this is happening, and how to resolve it.

**Comment by snibgo** [comment, marked as answer]
It occurs when the input has only one channel, which IM assumes is grayscale.

To fix this, insert "-set colorspace sRGB" immediately after the input filename.

**Reply by uriel-tannen** [reply]
Perfect, thanks @snibgo
